instrumentation/jms/src/main/java/brave/jms/PropertyFilter.java

```diff
@@ -40,28 +42,29 @@ static void filterProperties(Message message, Set<String> namesToClear) {
   }
 
   static void filterProperties(Message message, Set<String> namesToClear, List<Object> out) {
-    Enumeration<?> names;
+    List<String> names;
     try {
-      names = message.getPropertyNames();
+      names = list(message.getPropertyNames());
     } catch (Throwable t) {
       propagateIfFatal(t);
       log(t, "error getting property names from {0}", message, null);
       return;
     }
-
-    while (names.hasMoreElements()) {
```

Comment:
why is this logic changed? was there a problem? Using collections.list will add allocate an array, and changing this logic makes it hard to tell what change is actually a problem or in focus.